### PR TITLE
nostr: use network-specific CoinSwap event kinds for publish/discovery

### PR DESCRIPTION
**Pull request by Supremesv715**
# Pull Request

## Description
Adds different nostr kind events 
## Related Issue(s)
Closes #

Fixes #819 

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ x ] Code refactor / performance improvement
- [ ] Documentation update only
- [ ] CI / Docker / Build changes
- [ ] Other (please describe):

## Protocol Version(s) Affected
- [ ] Legacy (ECDSA — `messages.rs`, `contract.rs`, `handlers.rs`)
- [ ] Taproot-Musig2 (`messages2.rs`, `contract2.rs`, `handlers2.rs`)
- [x ] Both
- [ ] Neither (infrastructure / tooling only)

> **Note:** When modifying protocol logic, changes must be made to **both** versions unless the
> scope is explicitly version-specific.

## Affected Component(s)
- [ ] **makerd** (background daemon)
- [ ] **taker** (CLI client)
- [ ] **maker-cli** (command-line tool)
- [ ] Core protocol / cryptography / library
- [ x] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [ ] Tests / test framework
- [ ] Documentation (`docs/`)
- [ ] Other (please specify):

### Code Quality
- [ ]x I ran `cargo +nightly fmt --all` and committed the result
- [ x] I ran `cargo +stable clippy --all-features --lib --bins --tests -- -D warnings` with zero warnings
- [x ] I ran `cargo +stable clippy --examples -- -D warnings` with zero warnings
- [x ] I ran `RUSTDOCFLAGS="-D warnings" cargo +nightly doc --all-features --document-private-items --no-deps` with zero warnings
- [ x] Pre-commit git hook passes (`ln -s ../../git_hooks/pre-commit .git/hooks/pre-commit` if not already set)

### Testing
- [x ] All unit tests pass (`cargo test`)
- [ x] Integration tests pass (`cargo test --features integration-test`)
- [ x] Changes were manually tested on **regtest**
- [ x] End-to-end maker ↔ taker swap tested (where applicable)
- [ ] Test-only code is gated behind `#[cfg(feature = "integration-test")]`

### Documentation
- [ ] Relevant files in the `docs/` folder were updated
- [ ] Complex logic is commented

### Security & Privacy (Critical)
- [ ] This change preserves **trustlessness** and **atomic swap guarantees**
- [ ] No regression in **sybil resistance** or **fidelity bond** logic
- [ ] Tor anonymity and P2P message flow were reviewed
- [ ] No new attack vectors or trust assumptions introduced
- [ ] Edge cases and error handling considered
- [ ] ZMQ subscription integrity (raw tx/block feed) is unaffected
- [ ] `integration-test` feature flag is not reachable in production code paths

## How to Test

```bash
# Standard swap (good baseline)
cargo test --test standard_swap --features integration-test -- --nocapture

# Abort scenarios
cargo test --test abort1 --features integration-test -- --nocapture
cargo test --test taproot_taker_abort1 --features integration-test -- --nocapture

# Malicious behavior & recovery
cargo test --test malice1 --features integration-test -- --nocapture
cargo test --test taproot_timelock_recovery --features integration-test -- --nocapture

# Or run the full suite
cargo test --features integration-test
```

## Summary by CodeRabbit

* **Refactor**
  * Nostr bond broadcasts and subscriptions now derive event kind from the detected Bitcoin network (Main, Testnet, Signet, Regtest, Testnet4).
  * Watch tower discovery auto-detects the node's Bitcoin network and receives network context.

* **Behavior**
  * Watcher now fails early if the node reports an unsupported/unknown chain.

* **Chore**
  * Minor code cleanup in port-retry logic (no runtime behavior changes).

### CODE REVIEW

**Comment by mojoX911** [comment]
@stark-3k pls  take a look at this. Mostly it looks good to me. We can't consolidate it further as most nostr stuffs are independent routines. If this looks good to you we can get it in.

**Review by stark-3k — APPROVED** [review]
utACK

**Comment by stark-3k** [comment]
@Supremesv715 can you sign the commits and push https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

**Comment by Supremesv715** [comment]
> @Supremesv715 can you sign the commits and push https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

Hey @stark-3k i have signed the commit

File: src/watch_tower/nostr_discovery.rs [hunk 1]
```diff
@@ -201,12 +212,13 @@ fn handle_relay_message(
...
+    network: Network,
     seen_txid: &Arc<Mutex<SeenTxids>>,
     initial_sync_complete: &Arc<AtomicBool>,
 ) -> Result<(), WatcherError> {
     match msg {
         RelayMessage::Event { event, .. } => {
-            if event.kind != Kind::Custom(COINSWAP_KIND) {
+            if event.kind != Kind::Custom(coinswap_kind(network)) {
                 return Ok(());
             }
```
[the remark this answers is not in the document]
Comment (mojoX911) [reply src/watch_tower/nostr_discovery.rs:224]: @Supremesv715 this is a good suggestion. We can actually just calculate kind once at the most top level function and just pass `Kind` as an argument instead of `network` in subsequent calls. That would be the right design choice here.

File: src/watch_tower/watcher.rs [hunk 2]
```diff
@@ -114,6 +114,23 @@ impl<R: Role> Watcher<R> {
...
+            "test" => Network::Testnet,
+            "testnet4" => Network::Testnet4,
+            "signet" => Network::Signet,
+            "regtest" => Network::Regtest,
+            unknown => {
+                return Err(WatcherError::General(format!(
+                    "Unsupported Bitcoin network from node: {unknown}"
+                )))
+            }
+        };
```
Comment (mojoX911) [inline src/watch_tower/watcher.rs:133]: Idea for a follow up PR. We should put this as an utility routine in `utill.rs` and use that everywhere such network conversion is needed.

Comment (stark-3k) [reply src/watch_tower/watcher.rs:133]: good idea

**Comment by Supremesv715** [comment]
Hey @mojoX911 i have added the change suggested by coderabbit, you can review it

**Review by mojoX911 — APPROVED** [review]
Ack
